models/topic.model.js

```diff
@@ -7,14 +7,25 @@ exports.retrieveTopics = () => {
 };
 
 exports.retrieveArticleById = (article_id) => {
-  return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
-  .then((result) => {
-    if (result.rows.length === 0) {
-      return Promise.reject({
-        status: 404,
-        msg: 'Article not found'
-      })
-    }
-    return result.rows[0];
-  })
-}
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: "Article not found",
+        });
+      }
+      return result.rows[0];
+    });
+};
+
+exports.retrieveArticles = () => {
+  return db
+    .query(
+      `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url FROM articles JOIN comments ON articles.article_id = comments.article_id ORDER BY articles.created_at DESC;`
+    )
+    .then((result) => {
+      return result.rows;
+    });
+};
```

Comment:
See my feedback/comment above about using a `COUNT` aggregate function in this statement, to do the comment counting for you, so you don't have to it yourself. Otherwise, a good statement so far which orders and sorts by correctly (at this stage, you may need make it more dynamic later on though....)